Add New Project: Predict Home Prices with Python & Linear Regression (#128)
codedex-io/projects: projects/predict-home-prices-with-python-and-linear-regression/linear-regression.mdx

If you ever wondered how “prediction” charts are created in Python, the odds are that 1 of these 4 tools was involved. This quad of tools is the base for data analysis and machine learning which makes it easy for beginners to unlock the power of Python in data science.

After launching Notebook on Anaconda, open a new project:

<img
    src="https://github.com/codedex-io/projects/blob/3bb055969781c1b39575a95f0e7e34d84fddfba9/projects/predict-home-prices-with-python-and-linear-regression/notebook.png"
    alt="Anaconda Notebook"
/>

##Getting Started

In this tutorial, we will be using a [data set](https://docs.google.com/spreadsheets/d/11IDKeCPA5jMSG5I_KNps4VfFVfNyG8eIizwnm7KLR4o/edit#gid=0) that compares house size and house prices of properties sold in Brooklyn's Dumbo neighborhood to predict the cost of houses based on size. 

To get started, let’s import the libraries. Remember that since you are using an online emulator there is no need for local installations. This is a simple setup for linear regression using the scikit-learn library.

> ```

Author: This will give the snippet code highlights for Python.

<img
    src="https://github.com/codedex-io/projects/blob/3bb055969781c1b39575a95f0e7e34d84fddfba9/projects/predict-home-prices-with-python-and-linear-regression/file.png"
    alt="Upload File on Anaconda"
/>

```

> #Load data from CSV file
> data = pd.read_csv('home_dataset.csv')
>  
> #Extract features and target variable
> house_sizes = data ['HouseSize'].values
> house_prizes = data['HousePrice'].values

Author: ```suggestion
# Load data from CSV file
data = pd.read_csv('home_dataset.csv')
 
# Extract features and target variable
house_sizes = data['HouseSize'].values
house_prizes = data['HousePrice'].values
```

```

> We are using `pd.read_csv` to read the CSV file containing 59 pairs of data points. Next, we are extracting the values corresponding to the features on the data sheet. In this case, the data from the column named “HouseSize” is being added to the variable ‘house_size’.

Author: ```suggestion
We are using `pd.read_csv()` to read the CSV file containing 59 pairs of data points. Next, we are extracting the values corresponding to the features on the data sheet. In this case, the data from the column named “HouseSize” is being added to the ‘house_size’ variable.
```

Let's now get started on plotting the data, via the `plt` object from Matplotlib. First, we use `plt.scatter()` to create a scatter plot of house prices vs. house size. Next, we assign a title and x- and y-axis labels before using `plt.show()` to display the graph.

> ```

Author: ```suggestion
```py
```

> #Visualize the data
> plt.scatter(house_sizes, house_prices, marker ='o', color ='blue')
> plt.title('House Prices vs. House Size')
> plt.xlabel('House Size (sq.ft)')
> plt.ylabel('House Price ($)')
> plt.show()

Author: ```suggestion
# Visualize the data
plt.scatter(house_sizes, house_prices, marker ='o', color ='blue')
plt.title('House Prices vs. House Size')
plt.xlabel('House Size (sq.ft)')
plt.ylabel('House Price ($)')
plt.show()
```

```

> ##Train test

Author: ```suggestion

## Train test
```

Now that we have extracted our data and plotted on a graph, it's time for the linear regression part.

A [train-test](https://en.wikipedia.org/wiki/Training,_validation,_and_test_data_sets) splits the generated data into training and testing sets using the `train_test_split()` function from Sklearn.

The train test split is a model validation procedure commonly used in predictive machine learning to simulate how a model will work with new/unknown data. It is commonly used with large data sets or when you need a quick estimate.

> ```

Author: ```suggestion
```py
```

> #Split the data into training and testing sets
> X_train, X_test, y_train, y_test = train_test_split(house_sizes, house_prices, test_size=0.2, random_state=42)
> 

Author: ```suggestion
# Split the data into training and testing sets
x_train, x_test, y_train, y_test = train_test_split(house_sizes, house_prices, test_size=0.2, random_state=42)
```

For the `_train` and `_test`, let's use lowercase letters for the x- and y-axes.

```

> The code above splits the data 80/20:

Author: ```suggestion

The code above splits the data 80/20:
```

- 80% of the data will be used to train the algorithm in `x_train` and `y_train`.
- The other 20% will be used for testing the algorithm in `x_test` and `y_test`.

In this next step, we are creating and training the model to make predictions with the following code:

> ```

Author: ```suggestion
```py
```

> # Split the data into training and testing sets
> X_train, X_test, y_train, y_test = train_test_split(house_sizes, house_prices, test_size=0.2, random_state=42)
> 
> # Reshape the data for NumPy
> X_train = X_train.reshape(-1, 1)
> X_test = X_test.reshape(-1, 1)
> 
> # Create and train the model
> model = LinearRegression()
> model.fit(X_train, y_train)

Author: ```suggestion
# Split the data into training and testing sets
x_train, x_test, y_train, y_test = train_test_split(house_sizes, house_prices, test_size=0.2, random_state=42)

# Reshape the data for NumPy
x_train = x_train.reshape(-1, 1)
x_test = x_test.reshape(-1, 1)

# Create and train the model
model = LinearRegression()
model.fit(x_train, y_train)
```

```

> After preparing our data for Numpy with `.reshape()`, we can create the linear regression model with `LinearRegression()` and train it using our `x_train` and `y_train`data.
> 
> ##Predicting
> The model we just trained will now be able to make predictions on the test set `x-test` and `y-test`. The results are stored in the `predictions` variable.
> 
> We are now going to plot our results from our predictive modeling: 

Author: ```suggestion

After preparing our data for Numpy with `.reshape()`, we can create the linear regression model with `LinearRegression()` and train it using our `x_train` and `y_train` data.

## Predicting
The model we just trained will now be able to make predictions on the test set `x-test` and `y-test`. The results are stored in the `predictions` variable.

We are now going to plot our results from our predictive modeling: 
```

> ```

Author: ```suggestion
```py
```

> # Predict prices for the test set
> predictions = model.predict(X_test)
> 
> # Visualize the predictions
> plt.scatter(X_test, y_test, marker='o', color='blue', label='Actual Prices')
> plt.plot(X_test, predictions, color='red', linewidth=2, label='Predicted Prices')
> plt.title('Dumbo Property Price Prediction with Linear Regression')
> plt.xlabel('House Size (sq.ft)')
> plt.ylabel('House Price (millions $)')
> plt.legend()
> plt.show()

Author: ```suggestion
# Predict prices for the test set
predictions = model.predict(x_test)

# Visualize the predictions
plt.scatter(x_test, y_test, marker='o', color='blue', label='Actual Prices')
plt.plot(x_test, predictions, color='red', linewidth=2, label='Predicted Prices')
plt.title('Dumbo Property Price Prediction with Linear Regression')
plt.xlabel('House Size (sq.ft)')
plt.ylabel('House Price (millions $)')
plt.legend()
plt.show()
```

```

> With our `plt` object, we use a scatter plot to visualize the actual and predicted prices and show the results with blue dots and a red regression line:
> 
> <img
>        src="https://github.com/codedex-io/projects/blob/3bb055969781c1b39575a95f0e7e34d84fddfba9/projects/predict-home-prices-with-python-and-linear-regression/graph.png"
>        alt="Final Graph"
> />

Author: ```suggestion

With our `plt` object, we use a scatter plot to visualize the actual and predicted prices and show the results with blue dots and a red regression line:

<img
  src="https://raw.githubusercontent.com/codedex-io/projects/main/projects/predict-home-prices-with-python-and-linear-regression/graph.png"
  alt="Final Graph"
/>
```

> ## Conclusion
> That’s it, you have now officially trained and visualized a predictive modeling algorithm.
> 
> Take what you learned an extra step, let’s gather some data and use linear regression to visualize it. [Kaggle](https://www.kaggle.com/) is a great place to find data sets that have already been proven but you can also import any data. Get creative and remember to format your data for easy access.

Author: ```suggestion
## Conclusion

That’s it, you have now officially trained and visualized a predictive modeling algorithm.

Take what you learned an extra step by gathering some data and using linear regression to visualize it. [Kaggle](https://www.kaggle.com/) is a great place to find data sets that have already been proven but you can also import any data. Get creative and remember to format your data for easy access.
```
A few things:

- Shouldn't "predictive modeling algorithm" be "predictive linear regression model"?
- Below the conclusion, can we include something like a "More Resources"  section?

---
title: Predict Home Prices with Python and Linear Regression

> author: Full Name

Author: ```suggestion
author: Katherin Solis
```

> uid: 

Author: Zai06xd7TDfDZgzx6MjUgvwYJe32

> uid: 

Author: ```suggestion
uid: Zai06xd7TDfDZgzx6MjUgvwYJe32
```

This is the `uid` associated with your Codédex account

datePublished: YYYY-MM-DD
description: Use Python to model linear regression of home prices.
header:
bannerImage:
tags:
  - intermediate
  - python
---

<BannerImage
  link=""
  description=""
  uid={false}
  cl="for-sidebar"
/>

# Predict Home Prices with Python and Linear Regression

<AuthorAvatar
  author_name="Katherin Solis"

>   author_avatar=""

Author: ```suggestion
  author_avatar="/images/projects/authors/katherin_solis.png"
```

  username="katherin"
  uid={false}
/>

<BannerImage
  link=""
  description=""
  uid={false}
/>

**Prerequisites:** Python fundamentals
**Versions** Anaconda Online
**Read Time:** 60 minutes

> ## Introduction

Author: ```suggestion

## Introduction
```

- 🔢 [NumPy](https://numpy.org) offers a robust foundation for numerical operations and data analysis.
- 📖[Pandas](https://pandas.pydata.org/) allows you to analyze, clean, explore, and manipulate data from different sources
- 📈 [Matplotlib](https://matplotlib.org) transforms your data into compelling visuals like 2D graphs and barcharts.
- 🤖 [Scikit-learn](https://scikit-learn.org), commonly known as Sklearn, provides a user-friendly interface for all kinds of machine learning.

If you ever wondered how “prediction” charts are created in Python, the odds are that 1 of these 4 tools was involved. This quad of tools is the base for data analysis and machine learning which makes it easy for beginners to unlock the power of Python in data science.

After launching Notebook on Anaconda, open a new project:

<img
    src="https://github.com/codedex-io/projects/blob/3bb055969781c1b39575a95f0e7e34d84fddfba9/projects/predict-home-prices-with-python-and-linear-regression/notebook.png"
    alt="Anaconda Notebook"
/>

> ##Getting Started

Author: ```suggestion
## Getting Started
```

```
#Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(house_sizes, house_prices, test_size=0.2, random_state=42)

```
The code above splits the data 80/20:

- 80% of the data will be used to train the algorithm in `x_train` and `y_train`.
- The other 20% will be used for testing the algorithm in `x_test` and `y_test`.

In this next step, we are creating and training the model to make predictions with the following code:

```
# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(house_sizes, house_prices, test_size=0.2, random_state=42)

# Reshape the data for NumPy
X_train = X_train.reshape(-1, 1)
X_test = X_test.reshape(-1, 1)

# Create and train the model
model = LinearRegression()
model.fit(X_train, y_train)
```

After preparing our data for Numpy with `.reshape()`, we can create the linear regression model with `LinearRegression()` and train it using our `x_train` and `y_train` data.

> ## Predicting
> The model we just trained will now be able to make predictions on the test set `x-test` and `y-test`. The results are stored in the `predictions` variable.

Author: ```suggestion
## Predicting

The model we just trained will now be able to make predictions on the test set `x-test` and `y-test`. The results are stored in the `predictions` variable.
```

We are now going to plot our results from our predictive modeling: 

```py
# Predict prices for the test set
predictions = model.predict(x_test)

# Visualize the predictions
plt.scatter(x_test, y_test, marker='o', color='blue', label='Actual Prices')
plt.plot(x_test, predictions, color='red', linewidth=2, label='Predicted Prices')
plt.title('Dumbo Property Price Prediction with Linear Regression')
plt.xlabel('House Size (sq.ft)')
plt.ylabel('House Price (millions $)')
plt.legend()
plt.show()
```

With our `plt` object, we use a scatter plot to visualize the actual and predicted prices and show the results with blue dots and a red regression line:

<img
  src="https://raw.githubusercontent.com/codedex-io/projects/main/projects/predict-home-prices-with-python-and-linear-regression/graph.png"
  alt="Final Graph"
/>

> ## Conclusion
> That’s it, you have now officially trained and visualized a predictive modeling algorithm.

Author: ```suggestion
## Conclusion

That’s it, you have now officially trained and visualized a predictive modeling algorithm.
```

> To take your skills to the next level, consider sourcing data and utilizing linear regression for visualization.[Kaggle](https://www.kaggle.com/) is a great place to find data sets that have already been proven but you can also import any data. You can use linear regression to predict so many things:
> - Video game sales based on reviews.
> - Social media engagement based on follower growth.
> - Grades based on study time, and even recovery time based on medical data.
> 
> The possibilities are endless.
> Get creative and remember to format your data for easy access.

Author: ```suggestion
To take your skills to the next level, consider sourcing data and utilizing linear regression for visualization.[Kaggle](https://www.kaggle.com/) is a great place to find data sets that have already been proven but you can also import any data. You can use linear regression to predict so many things:

- Video game sales based on reviews.
- Social media engagement based on follower growth.
- Grades based on study time, and even recovery time based on medical data.

The possibilities are endless! Get creative and remember to format your data for easy access.
```

> [**More Resources:**]For additional learning and exploration, here are some resources to deepen your understanding:

Author: ```suggestion
### More Resources

For additional learning and exploration, here are some resources to deepen your understanding:
```
